server/emailHandler/emailService.js

```diff
@@ -0,0 +1,29 @@
+const nodemailer = require("nodemailer");
+
+// send email function - self explanatory
+const sendEmailFunction = (email, pincode) => {
+  var transporter = nodemailer.createTransport({
+    service: "gmail",
+    auth: {
+      user: `${process.env.EMAIL_USER}`,
+      pass: `${process.env.EMAIL_PASSWORD}`,
+    },
+  });
+
+  var mailOptions = {
+    from: "Your best friend",
+    to: `${email}`,
+    subject: "Hey ! this is your pin code for Caring",
+    text: `Hi there ! \n this is your pincode please insert it: \n ${pincode}`,
+  };
+
+  transporter.sendMail(mailOptions, function (error, info) {
+    if (error) {
+      res.status(404).send({ message: error.message });
+    }
```

Comment:
The function's responsibility is to send email, not to handle http requests. This violates the single responsibility principle.